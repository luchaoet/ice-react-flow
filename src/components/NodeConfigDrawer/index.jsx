import React from 'react';
import styles from './index.module.scss';
import { Drawer, Button, Message } from '@alifd/next';

const NodeConfigDrawer = (props) => {
  const { visible, currentNode, onClose } = props;
  const { name, Drawer: NodeDrawer, initProps = {} } = currentNode;

  function fieldValidate(error, data) {
    console.log('NodeConfigDrawer', error, data);
    if (error) {
      Message.error('请填写完整信息');
    }
  }

  return (
    <Drawer
      title={name}
      placement="right"
      width={400}
      visible={visible && !!NodeDrawer}
      onClose={() => onClose && onClose()}
    >
      <NodeDrawer
        initProps={initProps}
        fieldValidate={(error, data) => fieldValidate(error, data)}
      />
    </Drawer>
  );
};

export default NodeConfigDrawer;
