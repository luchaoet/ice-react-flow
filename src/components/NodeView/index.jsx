import React from 'react';
import styles from './index.module.scss';

function NodeView() {
  return <div className={styles.container}>NodeView</div>;
}

export default NodeView;
