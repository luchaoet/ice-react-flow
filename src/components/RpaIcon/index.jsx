import React from 'react';
import { Icon } from '@alifd/next';
import classnames from 'classnames';
import createdSVG from './font';
import styles from './index.module.scss';

if (process.env.NODE_ENV === 'test') {
  Icon.createFromIconfontCN({
    scriptUrl: 'http://at.alicdn.com/t/font_935323_9xvoh1907yh.js',
  });
} else {
  createdSVG();
}

function RpaIcon({ type, size = 'small' }) {
  const className = classnames('next-icon', styles['rpa-icon'], {
    [`next-${size}`]: size,
  });
  return (
    <i className={className}>
      <svg aria-hidden="true">
        <use xlinkHref={`#${type}`}></use>
      </svg>
    </i>
  );
}

export default RpaIcon;
