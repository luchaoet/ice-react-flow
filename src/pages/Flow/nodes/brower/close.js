export default {
  type: 'brower_close',
  title: '关闭网页',
  icon: { type: 'rpashubiaojianpan-keshihua' },
  canDraggable: true,
  canAfterAdd: true,
  canBeforeAdd: true,
  disabled: false,
  canSelect: true,
  description: '在浏览器中关闭网页',
  initProps: [
    {
      name: 'url',
      title: '网页地址',
      component: 'Input',
      defaultValue: 'www.baidu.com',
    },
  ],
};
