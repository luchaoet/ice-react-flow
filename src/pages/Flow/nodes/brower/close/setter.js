import React, { useEffect } from 'react'
import { Form, Input, Field, Radio } from '@alifd/next'
import IfElse, { If, Else } from '@/components/IfElse'
import ComponentInput from '@/components/ComponentInput'

const FormItem = Form.Item
const RadioGroup = Radio.Group

function Drawer({ initProps, fieldValidate: _fieldValidate }) {
  const field = Field.useField({
    values: initProps,
  })
  // const { getValue } = field

  function componentDidUnmount() {
    field.validate((error, data) => {
      _fieldValidate && _fieldValidate(error, data)
    })
  }

  useEffect(() => {
    field.validate()
    return componentDidUnmount
  }, [])

  return (
    <Form field={field}>
			
			<FormItem 
				label="" 
				required={true} 
				requiredMessage="请输入网页地址" 
				hasFeedback
			>
				<Input placeholder="" name="url" />
			</FormItem>
		
    </Form>
  )
}

export default Drawer
