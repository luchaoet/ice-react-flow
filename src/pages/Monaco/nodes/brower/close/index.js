import NodeView from '@/components/NodeView';
import Drawer from './setter';

export default {
  type: 'brower_close',
  name: '关闭网页',
  initProps: [{ key: '' }],
  NodeView,
  Drawer,
};
