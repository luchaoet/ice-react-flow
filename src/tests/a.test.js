// import React from 'react';
import renderer from 'react-test-renderer';
import NodeView from '../components/NodeView';
import RpaIcon from '@/components/RpaIcon';
import { shallow, mount } from 'enzyme';
import NodeConfigDrawer from '@/components/NodeConfigDrawer';
import close from '../pages/Flow/nodes/brower/close'

describe('图标测试', function() {
  it('renders', function() {
		const wrapper = mount(<RpaIcon type="iconkongjian-keshihua1" />);
    expect(wrapper.find('.next-icon').length).toBe(1);
  });
});

describe('节点测试', function() {
  it('renders', function() {
		const obj = close;
		expect(typeof obj.type).toEqual('string');
  });
});

describe('弹窗测试', function() {
  it('renders', function() {
		const wrapper = mount(
			<NodeConfigDrawer
        currentNode={close}
        visible={true}
      />
		)
		expect(wrapper.find('input').length).toBe(1);
  });
});


