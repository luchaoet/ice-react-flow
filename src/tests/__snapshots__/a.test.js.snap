// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render NodeView 1`] = `
<div
  className="container"
>
  NodeView
</div>
`;

exports[`render RpaIcon 1`] = `
<i
  className="next-icon next-medium"
  style={Object {}}
>
  <svg
    className="next-icon-remote"
    focusable={false}
  >
    <use
      xlinkHref="#iconkongjian-keshihua1"
    />
  </svg>
</i>
`;

exports[`renders 1`] = `
<div
  className="container"
>
  NodeView
</div>
`;

exports[`renders 2`] = `
<i
  className="next-icon next-medium"
  style={Object {}}
>
  <svg
    className="next-icon-remote"
    focusable={false}
  >
    <use
      xlinkHref="#iconkongjian-keshihua1"
    />
  </svg>
</i>
`;

exports[`renders NodeView 1`] = `
<div
  className="container"
>
  NodeView
</div>
`;

exports[`renders RpaIcon 1`] = `
<i
  className="next-icon next-medium"
  style={Object {}}
>
  <svg
    className="next-icon-remote"
    focusable={false}
  >
    <use
      xlinkHref="#iconkongjian-keshihua1"
    />
  </svg>
</i>
`;
